blockchain/ethereum.go

```diff
@@ -61,18 +58,24 @@ func newEVMClient(networkSettings EVMNetwork) (EVMClient, error) {
 		return nil, err
 	}
 
+	debugID++
 	ec := &EthereumClient{
 		NetworkConfig:       networkSettings,
+		debugID:             debugID,
 		Client:              cl,
-		NonceMu:             &sync.Mutex{},
 		Wallets:             make([]*EthereumWallet, 0),
-		Nonces:              make(map[string]uint64),
 		headerSubscriptions: map[string]HeaderEventSubscription{},
 		subscriptionMutex:   &sync.Mutex{},
 		queueTransactions:   false,
 		doneChan:            make(chan struct{}),
 	}
 
+	if ec.NetworkConfig.Simulated {
+		ec.NonceSettings = newNonceSettings()
```

Comment:
In case of load tests simulated network will also have parallel requests. Can we introduce it for simulated as well?

Reply:
If I want to send transactions in parallel in simulated chain using same owner address, will it help in that case? I am trying to figure out if it can replace the need for -  [ConcurrentEVMClient](https://github.com/smartcontractkit/chainlink-testing-framework/blob/1e9b4179660429e489e737eae1d0c3d0efb9d0ee/blockchain/ethereum.go#L706-L709)

Reply:
Yes, I think this can help in your case. Will make a ticket to incorporate it for that situation.